src/components/SearchBox/SearchBox.tsx

```diff
@@ -1,119 +1,105 @@
 import classNames from "classnames";
 import React from "react";
-import type { HTMLProps, FormEvent } from "react";
 
 import Icon from "../Icon";
 
-import type { ClassName, PropsWithSpread } from "types";
+import type { ClassName } from "types";
 
-export type Props = PropsWithSpread<
-  {
-    /**
-     * Whether autocomplete should be enabled for the search input.
-     */
-    autocomplete?: "on" | "off";
-    /**
-     * Optional classes to pass to the form element.
-     */
-    className?: ClassName;
-    /**
-     * Whether the input and buttons should be disabled.
-     */
-    disabled?: boolean;
-    /**
-     * Whether the input value will be controlled via external state.
-     */
-    externallyControlled?: boolean;
-    /**
-     * A function that will be called when the input value changes.
-     */
-    onChange?: (inputValue: string) => void;
-    /**
-     * A function that will be called when the form is submitted.
-     */
-    onSubmit?: (evt: FormEvent) => void;
-    /**
-     * A search input placeholder message.
-     */
-    placeholder?: string;
-    /**
-     * The value of the search input when the state is externally controlled.
-     */
-    value?: string;
-  },
-  HTMLProps<HTMLFormElement>
->;
+export type Props = {
+  /**
+   * Whether autocomplete should be enabled for the search input.
+   */
+  autocomplete?: "on" | "off";
+  /**
+   * Optional classes to pass to the form element.
+   */
+  className?: ClassName;
+  /**
+   * Whether the input and buttons should be disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Whether the input value will be controlled via external state.
+   */
+  externallyControlled?: boolean;
+  /**
+   * A function that will be called when the input value changes.
+   */
+  onChange?: (inputValue: string) => void;
+  /**
+   * A function that is called when the user clicks the search icon
+   */
+  onSearch: () => void;
```

Comment:
`onSearch` is marked as required here, is this expected?

If so, most of examples in storybook don't have it defined.

Reply:
I've made `onSearch` prop optional, I thought it should be required, but maybe it's not necessary.